### DISCUSSION TITLE
Snaps DApp Side Interface Questions

### DISCUSSION THREAD

**Discussion by cryptoKevinL** [MetaMask/snaps#1489] [Q&A] [answered]
As we develop our dApp to accommodate Snaps and MM Flask, we have some questions:

1) For the upcoming rollout of Snaps - will the main version of Metamask Support include Flask/Snaps?  This is a bit crucial as we either need to ask users in our dApp to update the existing install, or install a separate version of MM (understanding that having two versions in the same browser profile does not work).  

2) When a Snap is installed by the user, after the main rollout will the Flask-enabled Metamask auto update users Snaps if they opt-in to auto-update Metamask?  Or will there be a way within MM to let users know a new version of our Snap exists?

**Comment by cryptoKevinL** [comment]
@Montoya - any official word on these few items?  Will help us write up our documentation

**Comment by Montoya** [comment, marked as answer]
1. In September the MetaMask extension (not just Flask) will support the snaps on our allowlist. Any user will be able to install an allowlisted snap on the MetaMask extension (the one with tens of millions monthly active users). 

2. There is no auto-update for snaps yet. This is something we plan to support in the future, but for now, a user would have to go to your dapp to install a newer version. You could send a notification to users telling them there's a new version to update, but there's no way to actually deliver the update automatically.

**Comment by cryptoKevinL** [comment]
Thanks!  This helps a lot.
